internal/connection/connection.go

```diff
@@ -39,11 +43,29 @@ type Connection struct {
 
 // NewConnection establishes connection with sidecar, fetches capability and returns Connection object
 // filled with required information.
-func NewConnection(ctx context.Context, endpoint, nodeID, driverName, namespace, podName string) (*Connection, error) {
+func NewConnection(ctx context.Context, endpoint, nodeID, driverName, namespace, podName string, enableTLS bool) (*Connection, error) {
 	opts := []grpc.DialOption{
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
 		grpc.WithIdleTimeout(time.Duration(0)),
 	}
+
+	opts = append(opts, token.WithServiceAccountToken())
+	if enableTLS {
+
+		caFile, caError := token.GetCACert()
+		if caError != nil {
+			return nil, (fmt.Errorf("failed to get server cert %v", caError))
+		}
+		caCertPool := x509.NewCertPool()
+		if !caCertPool.AppendCertsFromPEM([]byte(caFile)) {
+			return nil, fmt.Errorf("failed to append CA cert")
+		}
+		tlsConfig := &tls.Config{
+			RootCAs:            caCertPool, // The CA certificates to verify the server
+			InsecureSkipVerify: true,
```

Comment:
dont we need to set it to true?

Reply:
By default setting it to false but making it configurable by users from CLI.

Reply:
After having discussions we need the ability to configure this to support certificates. We currently resolve the IP address and use that instead of host name. This causes certificate mismatch.